apps/prairielearn/src/pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.html.ts

```diff
@@ -0,0 +1,652 @@
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+
+import { assetPath, nodeModulesAssetPath } from '../../lib/assets';
+import {
+  AlternativeGroupSchema,
+  AssessmentQuestionSchema,
+  CourseInstanceSchema,
+  CourseSchema,
+  IdSchema,
+  QuestionSchema,
+  TopicSchema,
+  ZoneSchema,
+} from '../../lib/db-types';
+import { formatFloat } from '../../lib/format';
+import { STAT_DESCRIPTIONS } from '../shared/assessmentStatDescriptions';
+
+export const AssessmentQuestionStatsRowSchema = AssessmentQuestionSchema.extend({
+  course_short_name: CourseSchema.shape.short_name,
+  course_instance_short_name: CourseInstanceSchema.shape.short_name,
+  assessment_label: z.string(),
+  qid: QuestionSchema.shape.qid,
+  question_title: QuestionSchema.shape.title,
+  topic: TopicSchema,
+  question_id: IdSchema,
+  assessment_question_number: z.string(),
+  alternative_group_number: AlternativeGroupSchema.shape.number,
+  alternative_group_size: z.number(),
+  zone_title: ZoneSchema.shape.title,
+  start_new_zone: z.boolean(),
+  start_new_alternative_group: z.boolean(),
+});
+type AssessmentQuestionStatsRow = z.infer<typeof AssessmentQuestionStatsRowSchema>;
+
+export function InstructorAssessmentQuestionStatistics({
+  questionStatsCsvFilename,
+  rows,
+  resLocals,
+}: {
+  questionStatsCsvFilename: string;
+  rows: AssessmentQuestionStatsRow[];
+  resLocals: Record<string, any>;
+}) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head'); %>", resLocals)}
+        <script src="${nodeModulesAssetPath('lodash/lodash.min.js')}"></script>
+        <script src="${nodeModulesAssetPath('d3/dist/d3.min.js')}"></script>
+        <script src="${assetPath('localscripts/scatter.js')}"></script>
+        <script src="${assetPath('localscripts/histmini.js')}"></script>
+      </head>
+      <body>
+        <script>
+          $(function () {
+            $('[data-toggle="popover"]').popover({ sanitize: false });
+          });
+        </script>
+        ${renderEjs(__filename, "<%- include('../partials/navbar'); %>", resLocals)}
+        <main id="content" class="container-fluid">
+          ${renderEjs(
+            __filename,
+            "<%- include('../partials/assessmentSyncErrorsAndWarnings'); %>",
+            resLocals,
+          )}
+          ${resLocals.authz_data.has_course_permission_edit
+            ? html`
+                <div
+                  class="modal fade"
+                  id="refreshAssessmentQuestionStatsModal"
+                  tabindex="-1"
+                  role="dialog"
+                  aria-labelledby="refreshAssessmentQuestionStatsModalLabel"
+                >
```

Comment:
Do we want to use a `Modal` component here?

Reply:
Yes! Thanks for holding my PRs to the same standard as I hold yours 🙂 I'll get this changed.

Reply:
Changed in 1105ca6.